### PR TITLE
[IA-3676] [IA-3884] Add user input timeout for legacy and custom images.

### PR DESCRIPTION
**Pull request by n8landolt**
Ticket:
https://broadworkbench.atlassian.net/browse/IA-3676

Sub Task:
https://broadworkbench.atlassian.net/browse/IA-3884

Dependency is merged:
Depends on: https://broadworkbench.atlassian.net/browse/IA-3885 is complete.

This ticket adds a new const in tool-utils that defines our cached + supported imageIds. Using this list, we can show and hide the new NumberInput for users to input a custom timeout.
Screen Shot 2022-12-06 at 11 13 45 AM
Screen Shot 2022-12-06 at 11 19 05 AM

### CODE REVIEW

File: src/pages/workspaces/workspace/analysis/modals/ComputeModal.js [hunk 1]
```diff
@@ -1530,6 +1563,8 @@ export const ComputeModalBase = ({
           [Utils.DEFAULT, () => runtimeTypes.gceVm]
         )
         setSelectedLeoImage(value)
+        //TODO: Is this the best way to "un-set"?
+        setTimeoutInMinutes(supportedImages.includes(value) ? undefined : timeoutInMinutes)
```
Comment (msilva-broad) [inline src/pages/workspaces/workspace/analysis/modals/ComputeModal.js]: null is a better 'not set' value in my opinion... more explicit

File: src/pages/workspaces/workspace/analysis/modals/ComputeModal.test.js [hunk 2]
```diff
@@ -742,6 +742,9 @@ describe('ComputeModal', () => {
       const customImageUri = 'us'
       await fireEvent.change(imageInput, { target: { value: customImageUri } })
 
+      const customImageTimeoutLimit = await screen.findByText('Creation Timeout Limit')
+      expect(customImageTimeoutLimit).toBeInTheDocument()
```
Comment (msilva-broad) [inline src/pages/workspaces/workspace/analysis/modals/ComputeModal.test.js]: findByText will throw if not exactly one instance of text is found.  So this toBeInTheDocument check is not needed.  Throwing to fail the test is sufficient, and has the benefit of outputting helpful DOM debug info

File: src/pages/workspaces/workspace/analysis/modals/ComputeModal.js [hunk 3]
```diff
@@ -857,6 +862,34 @@ export const ComputeModalBase = ({
...
   }
 
+  const renderCustomTimeoutInMinutes = () => {
+    return h(IdContainer, [
+      id => h(div, {}, [
+        div({ style: { margin: '0.5rem 0' } }, [
+          label({ htmlFor: id, style: { ...computeStyles.label } }, ['Creation Timeout Limit']),
+          h(InfoBox, { style: { marginLeft: '0.5rem' } }, [
+            'Custom and Legacy image creation may take longer than the default 10 minute timeout window. ' +
+            'To avoid an error, you may enter a value between 10 and 45 minutes.'
```
Comment (LizBaldo) [inline src/pages/workspaces/workspace/analysis/modals/ComputeModal.js]: FYI, I think that @Qi77Qi mentioned that it would time out after 30 minutes anyway, so maybe replace 45 by 30 here?
I put a maximum of 30 minutes in the swagger doc so we should be consistent.

Comment (Qi77Qi) [reply src/pages/workspaces/workspace/analysis/modals/ComputeModal.js]: yea, taht's correct..we could change it to higher if needed in leo side

**Review by LizBaldo — APPROVED** [review]
LGTM! Thanks :)

File: src/pages/workspaces/workspace/analysis/modals/ComputeModal.test.js [hunk 4]
```diff
@@ -840,4 +842,151 @@ describe('ComputeModal', () => {
...
+
+        await userEvent.click(screen.getByText('Customize'))
+        const selectMenu = await screen.getByLabelText('Application configuration')
+        await userEvent.click(selectMenu)
+        const selectOption = await screen.findByText('Custom Environment')
+        await userEvent.click(selectOption)
+
+        await screen.findByText('Creation Timeout Limit')
+        const timeoutInput = await screen.getByLabelText('Creation Timeout Limit')
+        await fireEvent.change(timeoutInput, { target: { value: 20 } })
```
Comment (mlilynolting) [inline src/pages/workspaces/workspace/analysis/modals/ComputeModal.test.js:975]: Doesn't this test need to be changed to send no value, since this test case is meant to check null timeout submission?

**Review by mlilynolting — CHANGES_REQUESTED** [review]

File: src/pages/workspaces/workspace/analysis/modals/ComputeModal.test.js (hunk 4 above)
Comment (n8landolt) [reply src/pages/workspaces/workspace/analysis/modals/ComputeModal.test.js:975]: Maybe I should leave a comment here, add a new test, or rename this test.

What this line and selectOption2 is verifying that when the image is set to one of our "supported" images, the timeoutInMinutes will be null. 

See: 
https://github.com/DataBiosphere/terra-ui/pull/3560/files/824fb7b477f3f3ac21b1dab8157ea55201ef18b8#diff-a15b61bc748955fee99cca754569a25705e76701992dfc9da20bafbd7dffb664R1580

Comment (n8landolt) [reply src/pages/workspaces/workspace/analysis/modals/ComputeModal.test.js:975]: Look it over again and let me know if it clearer!

Comment (mlilynolting) [reply src/pages/workspaces/workspace/analysis/modals/ComputeModal.test.js:975]: OH yes this is clear! Thanks!

**Review by mlilynolting — APPROVED** [review]

**Comment by n8landolt** [comment]
DO NOT MERGE UNTIL

https://github.com/DataBiosphere/leonardo/pull/3064
OR
https://github.com/DataBiosphere/leonardo/pull/3078

Is merged and tested with this branch.

**Comment by Qi77Qi** [comment]
can we add metrics to record usage of this option...jsut a simple counter will do

**Comment by n8landolt** [comment]
We decided metrics would wait for the separate ticket! 
https://broadworkbench.atlassian.net/browse/IA-3902
